wallets/provider-walletconnect-2/src/session.ts

```diff
@@ -294,25 +240,24 @@ export function getAccountsFromSession(
        * It will return the chain id itslef if it's not that specific ID.
        */
       const chain = solanaChainIdToNetworkName(chainId.reference);
+      const isEvm = isEvmAddress(address);
+      if (isEvm) {
```

Comment:
Let keep this logic out of this function. `getAccountsFromSession` should return all the accounts from session, if you need only one evm account, you can make a separate function and filter the result whenever you need. (not in this function)